src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -28,92 +31,143 @@ public ToolManifestFinder(DirectoryPath probStart, IFileSystem fileSystem = null
-                        }
+        private SerializableLocalToolsManifest DeserializableLocalToolsManifest(FilePath possibleManifest)
+        {
+            return JsonConvert.DeserializeObject<SerializableLocalToolsManifest>(
+                _fileSystem.File.ReadAllText(possibleManifest.Value), new JsonSerializerSettings
+                {
+                    MissingMemberHandling = MissingMemberHandling.Ignore
+                });
+        }
 
-                        if (tools.Value.commands == null
-                            || (tools.Value.commands != null && tools.Value.commands.Length == 0))
-                        {
-                            packageLevelErrors.Add(LocalizableStrings.FieldCommandsIsMissing);
-                        }
+        private List<ToolManifestPackage> GetToolManifestPackageFromOneManifestFile(
+            SerializableLocalToolsManifest deserializedManifest, FilePath path)
+        {
+            List<ToolManifestPackage> result = new List<ToolManifestPackage>();
+            var errors = new List<string>();
 
-                        if (packageLevelErrors.Any())
-                        {
-                            var joined = string.Join(string.Empty, packageLevelErrors.Select(e => Environment.NewLine + "    " + e));
-                            errors.Add(string.Format(LocalizableStrings.InPackage, packageId.ToString()) + joined);
-                        }
-                        else
-                        {
-                            result.Add(new ToolManifestPackage(
-                                packageId,
-                                version,
-                                ToolCommandName.Convert(tools.Value.commands)));
-                        }
+            if (deserializedManifest.version == 0)
+            {
+                errors.Add(
+                    LocalizableStrings.ManifestVersion0 +
+                    path.Value);
+            }
+
+            if (deserializedManifest.version > SupportedVersion)
+            {
+                errors.Add(
+                    string.Format(
+                        LocalizableStrings.ManifestVersionHigherThanSupported,
+                        deserializedManifest.version, SupportedVersion) +
+                    path.Value);
+            }
+
+            foreach (KeyValuePair<string, SerializableLocalToolSinglePackage> tools in deserializedManifest.tools)
+            {
+                var packageLevelErrors = new List<string>();
+                var packageIdString = tools.Key;
+                var packageId = new PackageId(packageIdString);
+
+                string versionString = tools.Value.version;
+                NuGetVersion version = null;
+                if (versionString is null)
+                {
+                    packageLevelErrors.Add(LocalizableStrings.ToolMissingVersion);
+                }
+                else
+                {
+                    var versionParseResult = NuGetVersion.TryParse(
```

Comment:
You could collapse these two statements into one. Just put the TryParse in the body of the if condition.

Reply:
fixed, changed to one statement